### DISCUSSION TITLE
Is there a variable with the value of repo-url?

### DISCUSSION THREAD

**Discussion by ijmitch** [quarto-dev/quarto-cli#7820] [Q&A]
### Description

I would like to construct some links to, for example, milestones in my github repo and not repeat the repo's URL all over the place... something like:

`[the next milestone]({{< variable-for-repo-url >}}/milestone/1)`

but I can't see from the doc about variables what's available.

**Comment by cscheid** [comment]
If you use [metadata, then the shortcode makes those available](https://quarto.org/docs/authoring/variables.html#meta):

````qmd
---
key: value
format: html
---

This is the value of `key`: {{< meta key >}}.
````

<img width="242" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/285675/5a6b8806-f46a-4855-a3c2-822134665bdd">

**Reply by ijmitch** [reply]
Thanks for the suggestion. I was hoping that there might be a way to access the `repo-url` value from `_quarto.yml` rather than having to do this at the document level. 

I've done this in MkDocs via [config.repo_url](https://www.mkdocs.org/dev-guide/themes/#template-variables) available as a template variable.

**Reply by ijmitch** [reply]
Well, closest I've got is to use `_environment` with `repo="the-url"` but it would be nice to get at the one already in `_quarto.yml`.

Also surprised `_environment` isn't YAML.

**Reply by mcanouil** [reply]
Why `_environment` would be YAML? it's for environment variable. https://quarto.org/docs/projects/environment.html

**Reply by mcanouil** [reply]
You could use variables as described in https://quarto.org/docs/authoring/variables.html.

Unfortunately, not all fields accept shortcodes, so you cannot use the shortcode for repo-url.
<img width="1624" alt="Screenshot of a visual studio code window split in four parts. A '_variables.yml' file containing 'repo-url: URL', a '_quarto.yml' default file from a website quarto project where 'repo-url' is set using the 'var' shortcode to access '_variables.yml'. A 'index.qmd' which also uses the 'var' shortcode to get 'repo-url'. Finally, the HTML showing shortcodes replaced by the value of 'repo-url'" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/62770d5c-720f-4ff5-b6cf-e23683a5e7e9">

<img width="771" alt="A screenshot showing that shortcode set in 'repo-url' of '_quarto.yml' is not evaluated" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/0aa162c3-ba89-44fa-b829-4ccb8c042092">


@cscheid Would that be hard to make shortcode work in basically any field? (and not a too bad idea, especially regarding YAML validation)

**Reply by ijmitch** [reply]
OK, maybe I should have said having a way to structure some 'config' variables as an alternative to environment variables would be nice - so one could say something like `{{< config bob.alice >}}` like you can for `{{< meta bob.alice >}}`.

**Reply by mcanouil** [reply]
> You could use variables as described in https://quarto.org/docs/authoring/variables.html.

You mean variables? Why create another shortcode? What you call "config" seems to be exactly what "var" is.
(note that you could write a Lua shortcode extension)

**Reply by ijmitch** [reply]
oh! d'uh... YES!

**Reply by mcanouil** [reply]
Anyway, currently `repo-url` does not accept markdown, so as shown in the screenshot, you'll have to write the proper URL anyway in `website.repo-url`.

**Reply by ijmitch** [reply]
Yeah, I'm ok with having basically the same bit of info in `_quarto.yml` and `_variables.yml`.

**Reply by cscheid** [reply]
> @cscheid Would that be hard to make shortcode work in basically any field? (and not a too bad idea, especially regarding YAML validation)

It would for 1.4, because shortcode evaluation happens in Lua, but allowing it to expand in metadata means we would have to honor the expanded value in Typescript. And we use metadata values in Typescript often before we call Pandoc (where Lua execution happens)

**Reply by mcanouil** [reply]
Should I open an enhancement issue to track this for 1.5 or later then?
(it's kind of related with another issue about that not all `text` fields can evaluate shortcodes).

**Reply by cscheid** [reply]
> (it's kind of related with another issue about that not all text fields can evaluate shortcodes).

If we already have an issue open, would you mind locating that and adding a link to this discussion? Thanks.

**Reply by mcanouil** [reply]
The issue I was referring to is:
- #6359

For now the issue is really about `text` not other keys.
